go.mod

```diff
@@ -3,7 +3,6 @@ module github.com/DillonAd/d4bot
 go 1.20
 
 require (
-	github.com/spf13/viper v1.16.0
```

Comment:
Why remove this dependency?

Reply:
After switching to env vars, this dependency isn't in use anywhere that I can spot. Is this something that'd be useful during development?

Reply:
Why the switch to straight environment variables? That library tends to make the configuration pretty easy.

Reply:
Not that this needs to be a hardboiled production grade service, BUT in a production environment, env vars are usually pretty easy to feed to containers and configs are a bit harder. I'm assuming that we wouldn't want to build the config into the image, since a config change would necessitate a re-compile which is not the 12 factor way to do things. If you're shipping containers that need a config to run, you either have to store the config somewhere or synthesize it, then you have to volume mount it into the container. That's usually extra work when docker / k8s / helm natively let you pass env vars to containers.

Reply:
With viper the defaults are put into the `config.yaml` file, and then overridden with environment variables.
https://github.com/DillonAd/d4bot/blob/429161a346aa3f1d1220caf93b73791c959f2424/cmd/config/config.go#L28-L29
So I agree with you that we should use environment variables, I just had them set up through the `spf13/viper` library with defaults included. 

Parsing the configuration into a struct keeps the code terse and readable. If we add even five or ten more configurations then the code will become very repetitive with little to no value added.

The benefit of the library is to keep the configuration organized and readable as it grows.  The configuration items can be grouped into sub-structs under the main configuration so the purpose of each item remains as clear as possible.

Other reasons to use viper:
https://github.com/spf13/viper/blob/0520e750b595d6ca64603fce29ce0346d69a47c5/README.md?plain=1#L62-L86

( of course the formatting didn't work on that one.... (╯°□°）╯︵ ┻━┻ )

So in principle I agree with you, but I think I didn't make it clear how the library was being used.

Reply:
Per our conversation (would be very funny if anyone else on earth read this) this is now handled by `envconfig`. Thoughts welcome.